pkg/connector/collaborator_cache.go

```diff
@@ -100,50 +118,55 @@ func (p *collaboratorCache) buildCache(ctx context.Context) error {
 				for _, value := range values {
 					privilegeKey := workato.PrivilegeId(keyGroup, value)
 
-					p.privilegeToUser.Set(privilegeKey, compoundUser.Id(), compoundUser)
+					err = appendCachedValue[*CompoundUser](ctx, sessionStorage, privilegeToUserCachePrefix, privilegeKey, compoundUser)
+					if err != nil {
+						return fmt.Errorf("failed to set privilege to user cache in session storage: %w", err)
+					}
 				}
 			}
 
 			// Build for folders
 			for _, folderId := range collaboratorRole.FolderIDs {
-				p.folderToUser.Set(folderId, compoundUser.Id(), compoundUser)
+				folderIdStr := strconv.Itoa(folderId)
+				err = appendCachedValue[*CompoundUser](ctx, sessionStorage, folderToUserCachePrefix, folderIdStr, compoundUser)
+				if err != nil {
+					return fmt.Errorf("failed to set folder to user cache in session storage: %w", err)
+				}
 			}
 		}
 
 		// Build for roles
 		for _, role := range collaborator.Roles {
-			if role.EnvironmentType != p.env.String() {
+			if role.EnvironmentType != c.env.String() {
 				continue
 			}
 
-			p.roleToUser.Set(role.RoleName, compoundUser.Id(), compoundUser)
+			err = appendCachedValue[*CompoundUser](ctx, sessionStorage, roleToUserCachePrefix, role.RoleName, compoundUser)
+			if err != nil {
+				return fmt.Errorf("failed to set role to user cache in session storage: %w", err)
+			}
 		}
 	}
 
-	l.Info("Cache built for collaborators")
 	return nil
 }
 
-func (p *collaboratorCache) getUsersByPrivilege(privilegeKey string) []*CompoundUser {
-	if !p.initialized {
-		return nil
+func appendCachedValue[T any](ctx context.Context, sessionStorage sessions.SessionStore, cachePrefix string, cacheKey string, cacheValue T) error {
+	values, found, err := session.GetJSON[[]T](ctx, sessionStorage, cacheKey, sessions.WithPrefix(cachePrefix))
```

Comment:
I believe this does a network request to redis with every call and we should do a bulk set with SetManyJSON